src/components/Router/MainRouter.tsx

```diff
@@ -94,7 +95,20 @@ export default class MainRouter extends Component<InternalRouterProps, State> {
       steps && steps.some(isUploadFallbackOffAndShouldUseCamera)
     const { hasCamera } = this.props
 
-    return !isDesktop && !hasCamera && shouldStrictlyUseCamera === true
+    const findStep = buildStepFinder(steps)
+    const faceStep = findStep('face')
+
+    const photoCaptureFallback = faceStep?.options?.photoCaptureFallback ?? true
+
+    const canVideoFallbackToPhoto =
+      window.MediaRecorder != null || photoCaptureFallback
+
+    return (
+      !isDesktop &&
+      ((!hasCamera && shouldStrictlyUseCamera === true) ||
+        (!canVideoFallbackToPhoto &&
+          faceStep?.options?.requestedVariant === 'video'))
```

Comment:
I wonder if we can do better here, but not sure how to achieve it. For me, all of these brackets, `&&` and `||` are difficult to understand and maintain in the future. @stefaniacardenas @kopijunkie any suggestions?

Reply:
I agree that this can be hard to read. Not sure how I can make it clearer though. Maybe extracting everything after the `||` to another `const` helps? I'm open to all suggestions 😅

Reply:
Yeah, this is quite hard to read. We might be able to make lines 103-104 a util function since it's kind of duplicated in 2 different places/files. 🤔 

But lines 106 onwards is a tough one. Only thing I can think of is break them down into individual constants. 

Not sure if it might be worth creating a util function to check for the required options in the `faceStep` config.

Reply:
I'll try to make it more readable with those constants and build that util function to see how it looks :)